### PR TITLE
Document some internal SQL type ctors as additional-apis

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
## Summary

Documents internal ctors for three SQL-related types: `SqlGuid`, `SqlBinary`, and `SqlDecimal`. The out-of-band package _Microsoft.Data.SqlClient_ currently uses unsupported union overlays to access these types (see https://github.com/dotnet/runtime/issues/51836). We can update the code to use a less fragile private reflection pattern, but it first requires properly documenting these APIs under the _additional-apis_ subfolder.

### CODE REVIEW

File: docs/framework/additional-apis/system.data.sqltypes.sqlbinary.-ctor.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+api_name:
+  - "System.Data.SqlTypes.SqlBinary..ctor"
+api_location:
+  - "System.Data.dll"
+api_type:
+  - "Assembly"
+---
+# SqlBinary Constructor
+
+Initializes a new instance of the `SqlBinary` struct without making a copy of the `byte[]` passed via the _value_ parameter. Future mutations to this array may cause the resulting `SqlBinary` instance to change its value.
```
Comment (gewarren) [inline docs/framework/additional-apis/system.data.sqltypes.sqlbinary.-ctor.md:18]:
```suggestion
Initializes a new instance of the `SqlBinary` struct without making a copy of the `byte` array passed via the `value` parameter. Future mutations to this array may cause the resulting `SqlBinary` instance to change its value.
```

File: docs/framework/additional-apis/system.data.sqltypes.sqlbinary.-ctor.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+The _ignored_ parameter is ignored.
```
Comment (gewarren) [inline docs/framework/additional-apis/system.data.sqltypes.sqlbinary.-ctor.md:20]:
```suggestion
The `ignored` parameter is ignored.
```

File: docs/framework/additional-apis/system.data.sqltypes.sqlguid.-ctor.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+api_name:
+  - "System.Data.SqlTypes.SqlGuid..ctor"
+api_location:
+  - "System.Data.dll"
+api_type:
+  - "Assembly"
+---
+# SqlGuid Constructor
+
+Initializes a new instance of the `SqlGuid` struct without making a copy of the `byte[]` passed via the _value_ parameter. Future mutations to this array may cause the resulting `SqlGuid` instance to change its value.
```
Comment (gewarren) [inline docs/framework/additional-apis/system.data.sqltypes.sqlguid.-ctor.md:18]:
```suggestion
Initializes a new instance of the `SqlGuid` struct without making a copy of the `byte` array passed via the `value` parameter. Future mutations to this array may cause the resulting `SqlGuid` instance to change its value.
```

File: docs/framework/additional-apis/system.data.sqltypes.sqlguid.-ctor.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+The _ignored_ parameter is ignored.
```
Comment (gewarren) [inline docs/framework/additional-apis/system.data.sqltypes.sqlguid.-ctor.md:20]:
```suggestion
The `ignored` parameter is ignored.
```

File: docs/framework/additional-apis/system.data.sqltypes.sqlmoney.-ctor.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+api_name:
+  - "System.Data.SqlTypes.SqlMoney..ctor"
+api_location:
+  - "System.Data.dll"
+api_type:
+  - "Assembly"
+---
+# SqlMoney Constructor
+
+Initializes a new instance of the `SqlMoney` struct, where _value_ has already been scaled by a ten-thousandth of a currency unit. For example, if __20000__ is provided for the _value_ parameter, this `SqlMoney` instance will represent __2__ currency units.
```
Comment (gewarren) [inline docs/framework/additional-apis/system.data.sqltypes.sqlmoney.-ctor.md:18]:
```suggestion
Initializes a new instance of the `SqlMoney` struct, where `value` has already been scaled by a ten-thousandth of a currency unit. For example, if __20000__ is provided for the `value` parameter, this `SqlMoney` instance will represent __2__ currency units.
```

File: docs/framework/additional-apis/system.data.sqltypes.sqlmoney.-ctor.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+The _ignored_ parameter is ignored.
```
Comment (gewarren) [inline docs/framework/additional-apis/system.data.sqltypes.sqlmoney.-ctor.md:20]:
```suggestion
The `ignored` parameter is ignored.
```

**Review by gewarren — COMMENTED** [review]
We usually code fence parameter names so they don't get localized. Also, can you add the 3 new files to the TOC? Thank you!

**Comment by GrabYourPitchforks** [comment]
@gewarren Had a little bit of trouble finding the TOC file, but I think I got it now. :)
Let me know if anything else is missing! I appreciate your eagle eyes on this.

**Comment by GrabYourPitchforks** [comment]
Note: I used "structure" instead of "class" to match precedent already in the additional-apis section of the TOC.

```yml
  - name: stdole namespace
    items:
    - name: DISPPARAMS structure
      href: additional-apis/stdole.dispparams.md
    - name: EXCEPINFO structure
      href: additional-apis/stdole.excepinfo.md
```

**Comment by gewarren** [comment]
> @gewarren Had a little bit of trouble finding the TOC file, but I think I got it now. :) 

Sorry, I should have pointed you to it. It can be tricky to find.

**Review by gewarren — APPROVED** [review]

**Review by David-Engel — APPROVED** [review]
LGTM

**Pull request merged**